drools/src/main/java/com/datatorrent/drools/operator/StatelessDroolsOperator.java

```diff
@@ -256,19 +318,24 @@ public void afterRuleFlowGroupDeactivated(RuleFlowGroupDeactivatedEvent ruleFlow
     /**
      * This method will be called when a fact is inserted in the session.
      * This method will add the inserted fact in factsAndFiredRules Map.
+     * When the getRule() will not be null in that case the fact is added via a rule.
      *
      * @param objectInsertedEvent fact inserted in the session
      */
     @Override
     public void objectInserted(ObjectInsertedEvent objectInsertedEvent)
     {
-      factsAndFiredRules.put(objectInsertedEvent.getObject(), new ArrayList<Rule>());
+      Object newFact = objectInsertedEvent.getObject();
+      factsAndFiredRules.put(newFact, new ArrayList<Rule>());
```

Comment:
For every fact getting emitted on the factsOutput port need to have a corresponding entry here.